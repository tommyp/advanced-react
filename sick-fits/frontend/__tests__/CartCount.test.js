import { render, screen } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });

  it('Matches snapshot', () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });

  it('Updates via props', async () => {
    const { container, rerender, debug } = render(<CartCount count={12} />);
    expect(container.textContent).toBe('12');
    rerender(<CartCount count={13} />);
    expect(container.textContent).toBe('1312');
    await wait(400);
    expect(container.textContent).toBe('13');

    expect(container).toMatchSnapshot();
  });
});
