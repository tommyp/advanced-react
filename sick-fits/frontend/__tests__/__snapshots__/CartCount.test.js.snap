// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 hNdwpB"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 iephzA count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> Updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 hNdwpB"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 iephzA count count-enter-done"
      >
        13
      </div>
    </div>
  </span>
</div>
`;
